Add `Daily_challenge` (#12392)
ppy/osu-wiki: wiki/Daily_challenge/en.md

# Daily Challenge

> *Daily challenge* is a mode where the main task is to pass a selected beatmap that changes everyday sometimes with specific [mods](wiki/Gameplay/Game_modifier).

Reviewer: ```suggestion
The **daily challenge** is a multiplayer mode in osu!(lazer) where the main task is to pass a selected beatmap that changes everyday sometimes with specific [mods](wiki/Gameplay/Game_modifier).
```
For now, *multiplayer* is a bit stretched as a term. The devs planned to add more features later on though, with for example collaboratively achieving a certain total score.

## How to play

> *Notice: Daily challenge requires an [osu! account](/wiki/Registration), is not available for [silenced](/wiki/Silence) players and is only avaible in osu!(lazer).*

Reviewer: ```suggestion
*Notice: Daily challenge requires an [osu! account](/wiki/Registration), and is not available for [silenced](/wiki/Silence) players.*
```

Author: Maybe we should also say in the notice that dailies are only avaible for the osu! ruleset for now?
I feel it should be added since... well, it is only for osu! right now.

Author: Im not going to add this right away since i feel this needs a bit of a discussion since in the future it is planed to add dailies for every (main)  ruleset one by one

Reviewer: If we are going to add this, then I would say in the first paragraph.

Something along the lines of `for the osu! game mode` might work.

Author: applied the said suggestion, will leave this open for now regardless.

From the main menu, the daily challenge menu can be accessed with the following steps:

1. Click the `Play` button or press `P`.
2. Click the `Daily challenge` button or press `D`.

## Main Menu

![](img/Daily_challenge_menu.png "Example of a typical day in the daily challenge menu.") 

> On enter, the user is presented with an intro containing the beatmap the user has to pass, and the mods they have to pass with.

Reviewer: ```suggestion
On entering, the user is presented with an intro showing the beatmap the user has to pass, and the mods they have to pass it with.
```

> After the intro ends, the user can view the data for today's challenge and can interact with the chat that is currently only avaible in the daily challenge menu. 

Reviewer: ```suggestion
After the intro ends, the user can view the data for today's challenge. The left side displays various score-related information like the total pass count and the cumulative total score. In the middle, a leaderboard shows the top scores achieved by players. Users are able to discuss the daily challenge in the chat on the right.
```

## Selected Beatmaps

> Each day will introduce one new map from the featured artist lising with increased difficulty, resetting after a week.

Reviewer: ```suggestion
Each day will introduce one new map from the [featured artist](/wiki/People/Featured_Artists) listing with increased difficulty. Regularly after one week however, the difficulty would be reset again.
```
I'm not sure if including just the FA wiki article is enough here.

![](img/Beatmap_selection_criteria.png "Specific star ratings beatmaps have to follow for the entire week")

Sometimes, the maps could come with a mod forced onto users, which required them to pass the map with the selected mod.
## Trivia

### History

::: Infobox
![](img/yt_comment.jpg "waxxx14's comment suggesting to make a cup of the day format that can be found in")
:::

- The idea for daily challenge was taken from a comment from waxxx14 requesting a "cup of the day" format from [trackmania](https://en.wikipedia.org/wiki/TrackMania) in osu!, under a lazer dev video, ["deciding what to do with lazer"](https://www.youtube.com/watch?v=xUSxEjQQ1UI)

- The daily challenge launched to the public on July 25, 2024 in osu!(lazer) public relase 2024.725.0.

> - On relase, you couldn't use free mods, and you couldn't view the total pass count and the cumulative total score.

Reviewer: ```suggestion
- In the initial release, free mods were not selectable, and the total pass count as well as the cumulative total score were only added in a later update.
```

# Daily Challenge

The **daily challenge** is a multiplayer mode in osu!(lazer) where the main task is to pass a selected beatmap that changes everyday sometimes with specific [mods](/wiki/Gameplay/Game_modifier).

## How to play

*Notice: Daily challenge requires an [osu! account](/wiki/Registration), and is not available for [silenced](/wiki/Silence) players.*

From the main menu, the daily challenge menu can be accessed with the following steps:

1. Click the `Play` button or press `P`.
2. Click the `Daily challenge` button or press `D`.

## Main Menu

> ![](img/Daily_challenge_menu.png "Example of a typical day in the daily challenge menu.") 

Author: I had to change the screenshot because the resolution and file size didn't meet contributing guidelines, but thanks for showing me how to write this.

# Daily Challenge

> The **daily challenge** is a multiplayer mode in osu!(lazer) where players are able to build up a *daily* streak by passing beatmaps consecutively with increasing difficulty.

Reviewer 2: ```suggestion
The **daily challenge** is a multiplayer mode in osu!(lazer) where players are able to build up a *daily* streak by passing one predetermined beatmap each day.
```
The increase in difficulty in mentioned in line 5 so I think it's better to avoid a repetition.
Only stating "by passing beatmaps consecutively" one might be mislead into believing that it's a marathon where you need to play multiple maps in a row. The proposed change should clarify that it's one map per day for the streak

Reviewer: I would rather remove the repetition in the line below. Keep in mind, the most important information should come first.

The text mentioning that the difficulty resets each 7 days *may* be moved to the first paragraph as well.

Author: Went with Cookie's suggestion for now, will leave open for discussion.

> Each day will introduce one new map from the [featured artist](/wiki/People/Featured_Artists) listing with increased difficulty. Regularly after one week however, the difficulty would be reset again.

Reviewer 2: ```suggestion
Each day introduces a new map from the [Featured Artist](/wiki/People/Featured_Artists) listing with increased difficulty thorughout each week.
```

> Each day will introduce one new map from the [featured artist](/wiki/People/Featured_Artists) listing with increased difficulty. Regularly after one week however, the difficulty would be reset again.

Author: Used Walavouchey's suggestion for now, will leave open for disscusion

> Sometimes, the maps could come with a mod forced onto users, which required them to pass the map with the selected mod.

Reviewer 2: ```suggestion
Sometimes, the maps will come with a mod forced onto the users, which required them to play the map with the selected mod.
```
Would definitely change pass to play since the goal of the daily challenge is not to pass each map but to get a score as high as possible.

> Sometimes, the maps could come with a mod forced onto users, which required them to pass the map with the selected mod.

Author: > Would definitely change pass to play since the goal of the daily challenge is not to pass each map but to get a score as high as possible.

Thats a secondary goal technically - To get your rewards (streaks) you need to **pass** the beatmap with any mods you desire (+ the one the map is forced with), getting a score thats high as possible is a goal not everyone will have, and is a very different for each map, whatever if that is because of the song, overall map, forced mod, difficulty and anything else that affects gameplay expirience for the user.

I can switch it to this suggestion (at the moment of writing im currently with Walavouchey's suggestion), but I will need more input from others to decide on the final version.

## Game menu

> *Notice: Daily challenge requires an [osu! account](/wiki/Registration), and is not available for [silenced](/wiki/Silence) players.*

Author: I made this article based on the one for the multiplayer mode, so I thought that I might as well put this in here too.
If that's what I need to do then I'll do it tho I don't see it being there wrong exactly

> *Notice: Daily challenge requires an [osu! account](/wiki/Registration), and is not available for [silenced](/wiki/Silence) players.*

Author: after some thinking, i will delete it with the new batch of changes

From the main menu, the daily challenge menu can be accessed with the following steps:

1. Click the `Play` button or press `P`.

> 2. Click the `Daily challenge` button or press `D`.

Reviewer 2: Not rly sure why the users would need a guide on how to find the daily challenge. The article already makes it clear that it's a lazer only mode so people won't look for it in stable. I don't think we need to tell them any further as the daily challenge can't be missed when you look in lazer and if it can, then the design in lazer should be changed since something like the daily challenge should not be missable.

Author: I based this article based on the `Multiplayer` one, so i decided to add this. I also kept this because i personaly didn't even know that the main menu has keyboard shortcuts and i discovered them thanks to the outdated `Multiplayer` article which has them. 

I don't think this mini guide needs to be deleted, but if more people will want this change then ill drop the mini guide

Reviewer: The keyboard shortcuts are a useful addition in my opinion.

![](img/Daily-challenge-menu.png "Screenshot of the daily challenge menu as of November 7, 2024.") 

> On entering, the user is presented with an intro showing the beatmap the user has to pass, and the mods they have to pass it with.

Reviewer 2: ```suggestion
On entering, the user is presented with an intro showing the beatmap the user has to play, and the mods they have to pass it with.
```
Sure, the player has to pass the map, however its a competition for the best score and not just about whether or not one can pass.

> On entering, the user is presented with an intro showing the beatmap the user has to pass, and the mods they have to pass it with.

Author: will leave open for disscusion

After the intro ends, the user can view the data for today's challenge. The left side displays various score-related information like the total pass count and the cumulative total score. In the middle, a leaderboard shows the top scores achieved by players. Users are able to discuss the daily challenge in the chat on the right.

## Beatmap difficulty range

![](img/Beatmap-selection-criteria.png "Specific star ratings beatmaps have to follow for the entire week")

## Streak milestones

From playing the daily challenge, you can reach different milestones that will affect the rank of the number representing your streaks in your user profile.

<!---  #### Rank breakdown 

Note:  I didn't see the Total and Week requirements in the reference or i didn't understand the code, so some help with this would be nice

> | Rank  | Total streak | Daily streak | Weekly streak |

Author: Yea, i guessed so, but I don't know how to do that so I didn't do it 
(also I don't have the values for Total and Weekly streaks yet, and I think that's more important if we're listing this)

I'll look though the contributing guides to check on how to do it

> | Rank  | Total streak | Daily streak | Weekly streak |

Author: Im thinking that instead of making the numbers the rank colour (i feel this would get tacky) we could make a 
"badge "  but users wouldn't be able to get them and they would be basically only a background with the rank name in its colour 
(i feel like this would be the simplest and the best way to do it for now, unless you have a better idea)
(after some time we can give them better badges and reward users with the higher ones once they reach the desired rank)
